client/desktop/window-handlers/external-links/index.js

```diff
@@ -48,18 +52,48 @@ function replaceInternalCalypsoUrl( url ) {
 module.exports = function ( mainWindow ) {
 	const webContents = mainWindow.webContents;
 
-	webContents.on( 'will-navigate', function ( event, url ) {
+	webContents.on( 'will-navigate', async function ( event, url ) {
 		const parsedUrl = new URL( url );
+		log.info( `Navigating to URL: '${ parsedUrl }'` );
 
 		for ( let x = 0; x < ALWAYS_OPEN_IN_APP.length; x++ ) {
 			const alwaysOpenUrl = new URL( ALWAYS_OPEN_IN_APP[ x ] );
 
 			if ( domainAndPathSame( parsedUrl, alwaysOpenUrl ) ) {
+				log.info( 'Domain and path same, returning...' );
 				return;
 			}
 		}
 
-		openInBrowser( event, url );
+		event.preventDefault();
+
+		if ( isWpAdminPostUrl( url ) ) {
+			log.info( `Prompting to navigate to WP-Admin URL: '${ url }'` );
+
+			const redirectDialogOptions = {
+				buttons: [ 'Confirm', 'Cancel' ],
+				title: platform.isWindows() ? 'WordPress.com' : 'Proceed in External Browser?',
+				message: 'Proceed in External Browser?',
+				detail:
+					`Please note: WordPress Desktop does not support the Classic Editor plugin.` +
+					`\n\n` +
+					`In addition, Single Sign On is required to use this feature if on a Business Plan.`,
```

Comment:
Note

- There can be non-Atomic sites on a Business plan (SSO not needed)
- Also applies to Jetpack connected sites with any plan (not just Business)
- It's referred to as "WordPress.com Secure Sign-On" in the [support docs](https://wordpress.com/support/wordpress-com-secure-sign-on-sso/)

maybe something like this?

> WordPress.com Secure Sign-on is required to use this feature, when applicable.

---

Or it might be better just to exclude this copy, since it's not immediately applicable when the issue is a plugin that prevents Gutenframe from loading?

Reply:
> Or it might be better just to exclude this copy, since it's not immediately applicable when the issue is a plugin that prevents Gutenframe from loading?

Good callout. I think you're right - we're probably better off not confusing users any further. Given that there is a mechanism in the Desktop app to trigger SSO when applicable, I think it's fine to simplify and leave this out. Amended in ec96634a8d92695c585df76ebcb7a3e95f90ccb6